build-tools/packages/build-cli/src/instructionalPromptWriter.ts

```diff
@@ -44,6 +44,59 @@ interface Section {
     cmd?: string;
 }
 
+/**
+ * Links to ADO pipeline for all release group
+ */
+
+export enum ADOPipelineLinks {
+    CLIENT = "https://dev.azure.com/fluidframework/internal/_build?definitionId=12",
+    SERVER = "https://dev.azure.com/fluidframework/internal/_build?definitionId=30",
+    BUILDTOOLS = "https://dev.azure.com/fluidframework/internal/_build?definitionId=14",
+    AZURE = "https://dev.azure.com/fluidframework/internal/_build?definitionId=85",
+    APIMARKDOWNDOCUMENTER = "https://dev.azure.com/fluidframework/internal/_build?definitionId=97",
+    BENCHMARK = "https://dev.azure.com/fluidframework/internal/_build?definitionId=96",
+    TESTTOOLS = "https://dev.azure.com/fluidframework/internal/_build?definitionId=13",
+    TINYLICIOUS = "https://dev.azure.com/fluidframework/internal/_build?definitionId=22",
+    BUILDCOMMON = "https://dev.azure.com/fluidframework/internal/_build?definitionId=3",
+    ESLINTCONFLIGFLUID = "https://dev.azure.com/fluidframework/internal/_build?definitionId=7",
+    COMMONDEFINITIONS = "https://dev.azure.com/fluidframework/internal/_build?definitionId=8",
+    COMMONUTILS = "https://dev.azure.com/fluidframework/internal/_build?definitionId=10",
+    PROTOCOLDEFINITIONS = "https://dev.azure.com/fluidframework/internal/_build?definitionId=67",
+}
+
+/**
+ *
+ * Returns ADO pipeline link for the releaseGroup
+ */
+export const mapADOLinks = (releaseGroup: string | undefined): string => {
+    const adoLink = releaseGroup === "client"
+            ? ADOPipelineLinks.CLIENT
+            : releaseGroup === "server"
+            ? ADOPipelineLinks.SERVER
+            : releaseGroup === "azure"
+            ? ADOPipelineLinks.AZURE
+            : releaseGroup === "build-tools"
+            ? ADOPipelineLinks.BUILDTOOLS
+            : releaseGroup === "@fluid-tools/api-markdown-documenter"
+            ? ADOPipelineLinks.APIMARKDOWNDOCUMENTER
+            : releaseGroup ===  "@fluid-tools/benchmark"
+            ? ADOPipelineLinks.BENCHMARK
+            : releaseGroup === "@fluidframework/test-tools"
+            ? ADOPipelineLinks.TESTTOOLS
+            : releaseGroup === "tinylicious"
+            ? ADOPipelineLinks.TINYLICIOUS
+            : releaseGroup === "@fluidframework/build-common"
+            ? ADOPipelineLinks.BUILDCOMMON
+            : releaseGroup === "@fluidframework/eslint-config-fluid"
+            ? ADOPipelineLinks.ESLINTCONFLIGFLUID
+            : releaseGroup === "@fluidframework/common-definitions"
+            ? ADOPipelineLinks.COMMONDEFINITIONS
+            : releaseGroup === "@fluidframework/common-utils"
+            ? ADOPipelineLinks.COMMONUTILS
+            : ADOPipelineLinks.PROTOCOLDEFINITIONS;
+    return adoLink;
```

Comment:
Instead of this, I recommend using the `isReleaseGroup` type guard, and then a `switch` on the release group (maybe the map lookup will work). For the other packages, when you have your lookup indexed by the package/release group name, then you remove a lot of the code.

So something like:

```
if isReleaseGroup(g) {
  // switch on release group
} else {
  // g is a package, so you can look up the entry in the map...
  return linkMap.get(g);
}
```

Reply:
Wouldn't a direct map lookup work instead of using `isReleaseGroup` type guard?

Reply:
I wasn't sure if you could have keys of different types in a Map. Looks like you can!